### PR TITLE
Create script to add GPL license notice in given files, update source files with GPL license notice

### LINKED ISSUE #312: Every source code file should have a copyleft license notice at the top
As per the GPL.

### PR DESCRIPTION
**Pull request by daviewales**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [ ] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only

Issues:
- Fixes #312 

Purpose:
- Add a script to add the GPL license notice to files piped through `stdin` or provided as arguments. e.g. from `ripgrep`. The script pulls the current copyright information from the LICENSE file.
- Update LICENSE with the current copyright information
- Update source files with missing license notices using the following pipeline:  
   ```
   rg -i --files-without-match -tc -tcpp -tpy -tcmake -tsh -tps -g '!engine/src/cmd/collide2' 'General Public|BSD License|Copyright' | python3 script/add_gpl_license.p
   ```
- What release is this for? 0.9.x
- Is there a project or milestone we should apply this to?

### CODE REVIEW

File: script/add_gpl_license.py
Comment (BenjamenMeyer) [inline script/add_gpl_license.py:43]: This probably should be:
```
Copyright (c) 2001-2002 Daniel Horn
Copyright (c) 2002-2019 pyramid3d and other Vega Strike Contributors
Copyright (c) 2019-2020 Stephen G. Tuggy, and other Vega Strike Contributors
```
And then we'll continue updating that third line.

**Review by BenjamenMeyer — APPROVED** [review]
A suggestion on the copyright line; though it'd be good if we could get some input on the dates from @danielrh  and @pyramid3d ; perhaps @Loki1950  if they're not available.

I'm not going to block on it...but let's certainly have a discussion.

File: script/add_gpl_license.py
Comment (daviewales) [reply script/add_gpl_license.py:43]: What if we update the LICENSE file, and I can update the script to pull the copyright notice from there?

Comment (daviewales) [reply script/add_gpl_license.py:43]: I've added a [commit](https://github.com/vegastrike/Vega-Strike-Engine-Source/pull/544/commits/4086ac2474df33b9ab1d8837329ee36c2bdc9c68) to update the LICENSE file.
I'll tweak the script to pull from that when I get a chance.

Comment (daviewales) [reply script/add_gpl_license.py]: I've updated the script, and re-applied it to the files missing the license. The script now pulls the copyright information directly from the LICENSE file.

**Comment by daviewales** [comment]
I've pushed a new version of the license text, applied to all the files missing it. 😄

File: script/add_gpl_license.py
Comment (stephengtuggy) [inline script/add_gpl_license.py:82]: I don't think `'.h.in'` should be included here. Other than that, things are looking pretty good so far.

**Review by stephengtuggy — COMMENTED** [review]
Nice job, @daviewales . I didn't review every single file, but overall, it's looking pretty solid.

File: script/add_gpl_license.py
Comment (stephengtuggy) [reply script/add_gpl_license.py:82]: Oh, I think I remember. This file is input for CMake, which uses it to generate config.h. So this file does belong in the repository, and does need a copyleft notice. So never mind.

**Review by BenjamenMeyer — APPROVED** [review]
LGTM - didn't review all of them again like I did yesterday.
It'd still be nice to get a review from @danielrh and/or @pyramid3d or @Loki1950  as their rep.

**Review by stephengtuggy — APPROVED** [review]
LGTM

**Review by Loki1950 — APPROVED** [review]
I did not check all of the touched files but it looks good

File: script/add_gpl_license.py
Comment (daviewales) [reply script/add_gpl_license.py:82]: I was using ripgrep's filetype filters to get the list of files to update. `.h.in` and `.txt` are both there for CMake files.

**Comment by daviewales** [comment]
🎉

**Comment by Loki1950** [comment]
Thx for the fix @daviewales

**Pull request merged**
